ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/InitiatorBase.java

```diff
@@ -0,0 +1,309 @@
+ */
+
+package org.apache.hadoop.hive.ql.txn.compactor;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.common.ValidReadTxnList;
+import org.apache.hadoop.hive.common.ValidTxnList;
+import org.apache.hadoop.hive.common.ValidWriteIdList;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.CompactionRequest;
+import org.apache.hadoop.hive.metastore.api.CompactionResponse;
+import org.apache.hadoop.hive.metastore.api.CompactionType;
+import org.apache.hadoop.hive.metastore.api.GetValidWriteIdsRequest;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.metastore.api.NoSuchTxnException;
+import org.apache.hadoop.hive.metastore.api.Partition;
+import org.apache.hadoop.hive.metastore.api.StorageDescriptor;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hadoop.hive.metastore.metrics.AcidMetricService;
+import org.apache.hadoop.hive.metastore.metrics.Metrics;
+import org.apache.hadoop.hive.metastore.metrics.MetricsConstants;
+import org.apache.hadoop.hive.metastore.txn.CompactionInfo;
+import org.apache.hadoop.hive.metastore.txn.TxnCommonUtils;
+import org.apache.hadoop.hive.metastore.txn.TxnUtils;
+import org.apache.hadoop.hive.metastore.utils.JavaUtils;
+import org.apache.hadoop.hive.ql.io.AcidDirectory;
+import org.apache.hadoop.hive.ql.io.AcidUtils;
+import org.apache.hadoop.hive.shims.HadoopShims;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hive.common.util.Ref;
+
+import java.io.IOException;
+import java.security.PrivilegedExceptionAction;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class InitiatorBase extends MetaStoreCompactorThread {
+
+  static final private String COMPACTOR_THRESHOLD_PREFIX = "compactorthreshold.";
+
+  private List<CompactionResponse> initiateCompactionForMultiplePartitions(Table table,
+      Map<String, Partition> partitions, CompactionRequest request) {
+    List<CompactionResponse> compactionResponses = new ArrayList<>();
+    partitions.entrySet().parallelStream().forEach(entry -> {
+      try {
+        StorageDescriptor sd = resolveStorageDescriptor(table, entry.getValue());
+        String runAs = TxnUtils.findUserToRunAs(sd.getLocation(), table, conf);
+        CompactionInfo ci =
+            new CompactionInfo(table.getDbName(), table.getTableName(), entry.getKey(), request.getType());
+        ci.initiatorId = request.getInitiatorId();
+        ci.orderByClause = request.getOrderByClause();
+        ci.initiatorVersion = request.getInitiatorVersion();
```

Comment:
`ci.poolName` could be set here instead of passing it in `scheduleCompactionIfRequired(...)` method. This would also require to set poolName in Initiator at `String poolName = getPoolName(ci, t);`

As a result poolName parameter could be removed from scheduleCompactionIfRequired as the information is already in CompactionInfo

Reply:
Made changes as recommended